src/toolkits/style_transfer/style_transfer_data_iterator.cpp

```diff
@@ -0,0 +1,123 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include "style_transfer_data_iterator.hpp"
+
+#include <core/data/image/io.hpp>
+#include <model_server/lib/image_util.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+using neural_net::image_annotation;
+using neural_net::labeled_image;
+using neural_net::shared_float_array;
+
+flex_image get_image(const flexible_type& image_feature) {
+  if (image_feature.get_type() == flex_type_enum::STRING) {
+    return read_image(image_feature, /* format_hint */ "");
+  } else {
+    return image_feature;
+  }
+}
+
+std::vector<flexible_type> get_style(const data_iterator::parameters& params) {
+  gl_sarray style_images = params.style[params.style_column_name];
+
+  if (style_images.dtype() == flex_type_enum::IMAGE) {
+    style_images =
+        style_images.apply(image_util::encode_image, flex_type_enum::IMAGE);
+  }
+
+  gl_sframe result({{params.style_column_name, style_images}});
```

Comment:
Why not just use `gl_sarray::range_iterator`, instead of iterating over an SFrame?

Reply:
It was an artifact left over from me testing things. I'll refactor to use `gl_sarray`

Reply:
Irrelevant now.